### PR TITLE
Adds Timber library 

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Address #663

### CODE REVIEW

File: collect_app/build.gradle [hunk 1]
```diff
@@ -109,6 +109,9 @@ android {
 
     lintOptions {
         abortOnError true
+        textReport true
```
Comment (lognaturel) [inline collect_app/build.gradle]: Why did lint options have to change?

**Review by lognaturel — COMMENTED** [review]
Does this play nice with proguard minification? Since there is no explicit tag set, I suspect not and that we'd end up with mangled class names in our remote crash logging.

File: collect_app/build.gradle (hunk 1 above)
Comment (shobhitagarwal1612) [reply collect_app/build.gradle]: I was checking out different options and forgot to revert it back.

Comment (lognaturel) [reply collect_app/build.gradle]: Ah, cool. We can consider those in another PR if you'd like.

**Comment by shobhitagarwal1612** [comment]
> Does this play nice with proguard minification? Since there is no explicit tag set, I suspect not and that we'd end up with mangled class names in our remote crash logging.

Timber automatically adds a TAG to the logs and that is the name of the class

File: collect_app/build.gradle (hunk 1 above)
Comment (shobhitagarwal1612) [reply collect_app/build.gradle]: I'll revert it back

**Comment by lognaturel** [comment]
Yes, and how does it do it? If it does something like `MyActivity.class.getName();`, it won't work with minification.

**Comment by lognaturel** [comment]
Perhaps tags themselves don't matter if they're only used in debug builds. The question is whether stack traces would have garbled names in production remote logs as [here](https://github.com/opendatakit/collect/issues/360#issuecomment-275310133). I guess not?

**Comment by shobhitagarwal1612** [comment]
I'm not very sure whether the tags will have garbled names or not. If it doesn't happens then it's fine otherwise we can just log the error message and the throwable object or explicitly provide TAG for the exception in case of production.
Moreover we can fully customize the actions to be taken for Production logging using the customTree

**Comment by shobhitagarwal1612** [comment]
@lognaturel Please review the changes

**Comment by lognaturel** [comment]
I would rather not merge in a fake tree and I'd like to make sure we understand how things will show up in Firebase Crash before we move ahead!

File: collect_app/src/main/java/org/odk/collect/android/tasks/DeleteInstancesTask.java [hunk 2]
```diff
@@ -63,10 +62,12 @@ protected Integer doInBackground(Long... params) {
                 deleted += wasDeleted;
 
                 if (wasDeleted > 0) {
-                    Collect.getInstance().getActivityLogger().logAction(this, "delete", deleteForm.toString());
+                    Collect.getInstance()
+                            .getActivityLogger().logAction(this, "delete", deleteForm.toString());
                 }
             } catch (Exception ex) {
-                Log.e(TAG, "Exception during delete of: " + param.toString() + " exception: " + ex.toString());
+                Timber.e(ex, "Exception during delete of: "
```
Comment (nribeka) [inline collect_app/src/main/java/org/odk/collect/android/tasks/DeleteInstancesTask.java:69]: Example of code that throw log trace to the black hole 🤣 

Not your fault @shobhitagarwal1612 though. And I think we will have separate PR to fix this kind of issue 👍

File: collect_app/build.gradle [hunk 3]
```diff
@@ -137,6 +137,7 @@ allprojects {
     }
 }
 
+
```
Comment (nribeka) [inline collect_app/build.gradle:140]: Before committing code, I usually do Cmd + K and go through the changes really quick to make sure there's no changes that I don't want gets committed. Sometimes I forget to that too ...

So revert this empty line.

**Comment by lognaturel** [comment]
Thanks for getting us started on this, @shobhitagarwal1612! To address this a little bit more incrementally, let's go for a pull request just adding the dependency and setting up a `Tree` for Firebase Crash. @Ishmeetsingh97 is interested in doing this. @Ishmeetsingh97, there are some examples online that you can look through. On the pull request, please make sure to describe what you do to verify that it's working. You'll need to add some logging calls to verify it but please remove them before you commit.

**Pull request closed without merging**